src/constraint/CollisionFree.cpp

```diff
@@ -42,16 +43,47 @@ bool CollisionFree::isSatisfied(
         groups.second.get(),
         mCollisionOptions,
         &collisionResult);
+
     if (collision)
+    {
+      if (_outcome != nullptr)
+      {
+        const auto& collisionOutcome
+            = dynamic_cast<CollisionFreeOutcome*>(_outcome);
+        if (collisionOutcome == nullptr)
+          throw std::invalid_argument(
+              "TestableOutcome pointer is not of type CollisionFreeOutcome.");
+        const auto& contacts = collisionResult.getContacts();
+        for (const auto& elem : contacts)
+        {
+          collisionOutcome->markPairwiseContact(elem);
+        }
+      }
       return false;
+    }
   }
 
   for (auto group : mGroupsToSelfCheck)
```

Comment:
Nit: This is irrelevant to this PR, but let's avoid unnecessary copy here by changing this line to `for (const auto& group : mGroupsToSelfCheck)`.

Also `for (auto groups : mGroupsToPairwiseCheck)` should be `for (const auto& groups : mGroupsToPairwiseCheck)`